File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode t=new ListNode(-1);
        ListNode t1=t;
        int c=0,x,s;
        while (l1!=null && l2!=null){
            x=c+l1.val+l2.val;
            s=x%10;
            c=x/10;
            t.next=new ListNode(s);
            t=t.next;
            l1=l1.next;
            l2=l2.next;
        }
        while(l1!=null){
            if(c!=0){
                x=c+l1.val;
                c=0;
                }
            else{
                x=l1.val;
                }
                s=x%10;
                c=x/10;
                t.next=new ListNode(s);
                t=t.next;
                l1=l1.next;
    }
    while(l2!=null){
            if(c!=0){
                x=c+l2.val;
                c=0;
                }
            else{
                x=l2.val;
                }
                s=x%10;
                c=x/10;
                t.next=new ListNode(s);
                t=t.next;
                l2=l2.next;
    }
    if(c!=0)
    t.next=new ListNode(c);
    return t1.next;
}
}

File: 0002-add-two-numbers/0002-add-two-numbers.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* addTwoNumbers(struct ListNode* l1, struct ListNode* l2)
{
    struct ListNode *l3=NULL;
    int s=0,c=0;
    while(l1!=NULL && l2!=NULL)
    {
        struct ListNode *new=malloc(sizeof(struct ListNode));
        s= (l1->val +l2->val +c)%10;
        
        c=(l1->val +l2->val +c)/10;
        new->val=s;
        if(l3==NULL)
        {
            l3=new;
            new->next=NULL;
        }
        else{
            struct ListNode *ptr=l3;
            while(ptr->next!=NULL)
            {
                ptr=ptr->next;
            }
            ptr->next=new;
            new->next=NULL;

        }
        l1=l1->next;
        l2=l2->next;
    }
    struct ListNode *rem;
    if (l1==NULL)
        rem=l2;
    else
        rem=l1;
    while(rem!=NULL)
    {
        struct ListNode *new=malloc(sizeof(struct ListNode));
        new->next=NULL;
        s=(rem->val +c)%10;
        c=(rem->val +c)/10;
        new->val=s;
        
        struct ListNode *ptr=l3;
            while(ptr->next!=NULL)
            {
                ptr=ptr->next;
            }
            ptr->next=new;
            new->next=NULL;            
        rem=rem->next;

    }
    if(c!=0)
    {
        struct ListNode *new=malloc(sizeof(struct ListNode));
        new->next=NULL;
        new->val=c;
        struct ListNode *ptr=l3;
            while(ptr->next!=NULL)
            {
                ptr=ptr->next;
            }
            ptr->next=new;
            new->next=NULL;     
    }
        return l3;
}